Mathlib/NumberTheory/LSeries/HurwitzZetaValues.lean

```diff
@@ -0,0 +1,197 @@
+/-
+Copyright (c) 2024 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+
+import Mathlib.NumberTheory.ZetaValues
+import Mathlib.NumberTheory.LSeries.HurwitzZeta
+
+/-!
+# Special values of Hurwitz zeta functions
+
+This file gives the formulae for values of Hurwitz zeta functions at negative integers in terms
+of Bernoulli polynomials.
+
+(Note that most of the actual work for these formulae is done elsewhere, in
+`Mathlib.NumberTheory.ZetaValues`. This file has only those results which really need the
+definition of Hurwitz zeta and related functions, rather than working directly with the defining
+sums in the convergence range.)
+
+## Main results
+
+- `hurwitzZeta_neg_nat`: for `k : ℕ` with `k ≠ 0`, and any `x ∈ ℝ / ℤ`, the special value
+  `hurwitzZeta x (-k)` is equal to `-(Polynomial.bernoulli (k + 1) x) / (k + 1)`.
+
+## TODO
+
+* Derive the special case of the Riemann zeta function from these results (rather than proving it
+  separately). This will come in a future PR which will re-implement Riemann zeta as a special
+  case of Hurwitz zeta.
+* Extend to cover Dirichlet L-functions.
+* The formulae are correct for `s = 0` as well, but we do not prove this case, since this requires
+  Fourier series which are only conditionally convergent, which is difficult to approach using the
+  methods in the library at the present time.
+-/
+
+open Complex Real Set
+
+open scoped Nat
+
+variable {k : ℕ} {x : ℝ}
+
+/-- Express the value of `cosZeta` at a positive even integer as a value
+of the Bernoulli polynomial. -/
+theorem cosZeta_two_mul_nat (hk : k ≠ 0) (hx : x ∈ Icc (0 : ℝ) 1) :
```

Comment:
This and most of the lemmas below express an "easy"  LHS in terms of a "complicated" RHS. I understand that this is closer to the mathematician's idea that we are "expressing the Hurwitz zeta in terms of..." but I wonder if the opposite would not be more natural given the library design. Do you already have applications showing that these directions are more useful than the opposite ones?

Reply:
That’s an Interesting point, but I think I disagree about which side is “complicated”. The RHS is lengthy to write down but all the terms are elementary in some sense (particularly for the lemmas with `s < 0`, where the RHS is a rational number and algorithmically computable, while the LHS takes literally thousands of lines of code to even define.) E.g. in “zeta(-1) = 1/12” I think it’s clear which side is the easy one!

Reply:
Yes, I see what you mean and I also agree that the direction you chose is the best one. I was more thinking about "is a mathematician's and a mathlib-ician's definition of "easy" the same one"? One can argue that "simple things" are goal states you are happy with, and "hard ones" are goal states you are unhappy with. If the above lemma is typically used to transform the RHS into a value of the cosZeta function, that it shuold be changed, but if it is meant to unwind coZeta(2k) than this is the right direction (same idea as for simp). For the code at hand I do not want to suggest modifications, after all, I was just opening the discussion ;-)

Reply:
Indeed, it's meant as a way to "unwind" cosZeta(2k) in terms of more elementary things (as a step towards formulae like zeta(2) = pi^2 / 6).